python/mujincontrollerclient/uriutils.py

```diff
@@ -586,9 +586,14 @@ def parseResult(self):
 
     @property
     def filename(self):
+        suffix = _EnsureUTF8(self._suffix)
+        if suffix and self._primaryKey.endswith(suffix):
+            basePartType = _Unquote(self._primaryKey[:-len(suffix)])
+        else:
+            basePartType = _Unquote(self._primaryKey)
         if not self._mujinPath:
-            return self.partType + self._suffix
-        return os.path.join(self._mujinPath, self.partType + self._suffix)
+            return basePartType + self._suffix
```

Comment:
This looks like too much dancing.

Reply:
i think he copied from here https://github.com/mujin/mujincontrollerclientpy/commit/4a888c79b7be985fa14046cf5af16117eeb582ff